db/revision_cache.go

```diff
@@ -150,32 +112,28 @@ func NewRevisionCacheShard(capacity uint32, loaderFunc RevisionCacheLoaderFunc,
 // Returns the body of the revision, its history, and the set of channels it's in.
 // If the cache has a loaderFunction, it will be called if the revision isn't in the cache;
 // any error returned by the loaderFunction will be returned from Get.
-func (rc *RevisionCache) Get(docid, revid string) (DocumentRevision, error) {
-	return rc.getFromCache(docid, revid, BodyShallowCopy, rc.loaderFunc != nil)
+func (rc *LRURevisionCache) Get(docID, revID string, copyType BodyCopyType) (DocumentRevision, error) {
+	return rc.getFromCache(docID, revID, copyType, rc.loaderFunc != nil)
 }
 
 // Looks up a revision from the cache only.  Will not fall back to loader function if not
 // present in the cache.
-func (rc *RevisionCache) GetCached(docid, revid string) (DocumentRevision, error) {
-	return rc.getFromCache(docid, revid, BodyShallowCopy, false)
-}
-
-// Returns the body of the revision based on the specified body copy policy (deep/shallow/none)
-func (rc *RevisionCache) GetWithCopy(docid, revid string, copyType BodyCopyType) (DocumentRevision, error) {
-	return rc.getFromCache(docid, revid, copyType, rc.loaderFunc != nil)
+func (rc *LRURevisionCache) Peek(docID, revID string) (docRev DocumentRevision, found bool) {
+	docRev, _ = rc.getFromCache(docID, revID, BodyShallowCopy, false)
```

Comment:
I agree with the change to make Peek not return error (unlike GetCached) - the fact that the current GetCached returns error is really a consequence of the underlying implementation.  However, I don't think we should ignore error here - if we get a non-nil error back from the getFromCache call, I think we should be returning found=false, regardless of whether body is nil.